util/compiler.h

```diff
@@ -5,11 +5,13 @@
 /* Use to tag a variable that causes compiler warnings. Use as:
     int uninitialized_var(sz)
 
-   This is only enabled for old compilers. gcc 6.x and beyond have excellent
-   static flow analysis. If code solicits a warning from 6.x it is almost
-   certainly too complex for a human to understand.
+   This is only enabled for old compilers and non-x86 architectures. gcc 6.x
+   and beyond have excellent static flow analysis for x86 platforms. If code
+   solicits a warning from 6.x it is almost certainly too complex for a human to
+   understand.
 */
-#if __GNUC__ >= 6 || defined(__clang__)
+#if (__GNUC__ >= 6 && !defined(__aarch64__)) || \
+    defined(__clang__)
 #define uninitialized_var(x) x
```

Comment:
@shamisp - certainly this case has been fixed for a long time in x86. It is probably worth referring the Linaro compiler team to this example.. It would be interesting to know of upstream gcc has this problem too.

I'd rather you use -Wno-maybe-uninitialized in the travis wrapper than globally modify compiler.h because I assume this will eventually get fixed in gcc.